src/react/Sidebar.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2021 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React, { useCallback, useMemo } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import { Nav, NavItem, NavLink } from 'react-bootstrap';
+import AstarteClient from 'astarte-client';
+
+import useFetch from './hooks/useFetch';
+import useInterval from './hooks/useInterval';
+
+interface SidebarApiStatusProps {
+  astarte: AstarteClient;
+}
+
+const SidebarApiStatus = ({ astarte }: SidebarApiStatusProps) => {
+  const appEngineHealth = useFetch(astarte.getAppengineHealth);
+  const realmManagementHealth = useFetch(astarte.getRealmManagementHealth);
+  const pairingHealth = useFetch(astarte.getPairingHealth);
+  const flowHealth = astarte.features.flow ? useFetch(astarte.getFlowHealth) : null;
+
+  const refreshHealth = useCallback(() => {
+    appEngineHealth.refresh();
+    realmManagementHealth.refresh();
+    pairingHealth.refresh();
+    if (flowHealth) {
+      flowHealth.refresh();
+    }
+  }, [appEngineHealth, realmManagementHealth, pairingHealth, flowHealth]);
+
+  useInterval(refreshHealth, 30000);
```

Comment:
this should be wrapped in an effect. We need to clear the interval when the sidebar is not shown (for instance, in the login page)

Reply:
It is indeed using the `useInterval` hook which clears its interval automatically